aten/src/ATen/native/cuda/SparseBinaryOpIntersectionKernel.cu

```diff
@@ -18,16 +18,23 @@ struct CUDAKernelLauncher {
 
 struct MulOp {
   template <typename scalar_t>
-  static FUNCAPI scalar_t apply(scalar_t a, scalar_t b) {
+  static FUNCAPI INLINE scalar_t apply(scalar_t a, scalar_t b) {
```

Comment:
It only kicks in for CUDA to make sure it is forceinlined.

Reply:
According to aten/src/ATen/native/sparse/Macros.h, `INLINE` is `inline` in non-CUDA context.

Perhaps this code is confusing because of the reason why `apply` is defined `static`: it is syntactic sugar that allows `MulOp::apply` expression without creating `MulOp` instance, that is, `struct` definition is used as a namespace. This leads to a question, why not use a namespace, to begin with? E.g.
```c++
namespace MulOp {
  template <typename scalar_t>
  FUNCAPI INLINE scalar_t apply(scalar_t a, scalar_t b) {...}
}
```

Reply:
Because I find it more eye-pleasing to have a struct and not a just a parametric function, plus a struct can be passed as a non-templated argument for other use cases.

Reply:
> According to aten/src/ATen/native/sparse/Macros.h, `INLINE` is `inline` in non-CUDA context.
It is used in a `*.cu` file, so is in the CUDA context here.